src/Bicep.Core/Emit/TemplateEmitter.cs

```diff
@@ -27,6 +27,35 @@ public TemplateEmitter(SemanticModel model, EmitterSettings settings)
         /// </summary>
         public static Encoding UTF8EncodingWithoutBom { get; } = new UTF8Encoding(false);
 
+        /// <summary>
+        /// Emits a template to the specified stream if there are no errors. No writes are made to the stream if there are compilation errors.
+        /// </summary>
+        /// <param name="stream">The stream to write the template</param>
+        public EmitResult EmitParametersFile(Stream stream) => EmitOrFail(() =>
+        {
+            // read content of the stream into a string
+            using var reader = new StreamReader(stream);
+            var content = reader.ReadToEndAsync().GetAwaiter().GetResult();
```

Comment:
GetResult() will block the thread, seems like you should use an async function with await.

Reply:
If it's difficult to implement, please ask @majastrz  or @anthony-c-martin.

Reply:
Use the sync version instead:

var content = reader.ReadToEnd();

Reply:
done! thanks...